scripts/connectivity_func_v_struc.py

```diff
@@ -0,0 +1,490 @@
+    os.makedirs(boxplot_dir, exist_ok=True)
+    for roi in rois:
+        all_results_roi = all_results[all_results["selected_rois"] == roi]
+        d = {"Comparison": [], "FC measure": [], "Correlation FC v SC": []}
+        for _, row in all_results_roi.iterrows():
+            n_subs = len(row["kept_subjects"])
+            corr = row["correlation"].reshape(n_subs, n_subs)
+            same_sub = np.diagonal(corr, offset=0).tolist()
+            upper_tri = corr[np.triu_indices_from(corr, k=1)].tolist()
+            lower_tri = corr[np.tril_indices_from(corr, k=-1)].tolist()
+            cross_sub = upper_tri + lower_tri
+            d["Comparison"].extend(
+                ["Within Subject"] * len(same_sub)
+                + ["Across Subject"] * len(cross_sub)
+            )
+            d["FC measure"].extend(
+                [row["task1_measure"]] * (len(same_sub) + len(cross_sub))
+            )
+            d["Correlation FC v SC"].extend(same_sub + cross_sub)
+            # p_values.append(row["p_value"])
+        df = pd.DataFrame(d)
+        p_values = []
+        for fc_measure in df["FC measure"].unique():
+            df_fc = df[df["FC measure"] == fc_measure]
+            same_sub = df_fc[df_fc["Comparison"] == "Within Subject"][
+                "Correlation FC v SC"
+            ].values
+            cross_sub = df_fc[df_fc["Comparison"] == "Across Subject"][
+                "Correlation FC v SC"
+            ].values
+            t_test = stats.ttest_ind(
+                same_sub, cross_sub, alternative="greater"
+            )
+            p_values.append(t_test[1])
+        fig, ax = plt.subplots()
+        sns.boxplot(
+            x="Correlation FC v SC",
+            y="FC measure",
+            hue="Comparison",
+            data=df,
+            ax=ax,
+            orient="h",
+            fliersize=0,
+        )
+        for i, p in enumerate(p_values):
+            if p > 0.05:
+                signif = "ns"
+            elif p <= 0.0001:
+                signif = "p ≤ 0.0001"
+            elif p <= 0.001:
+                signif = "p ≤ 0.001"
+            elif p <= 0.01:
+                signif = "p ≤ 0.01"
+            elif p <= 0.05:
+                signif = "p ≤ 0.05"
+            ax.text(
+                df["Correlation FC v SC"].min(),
+                i,
+                signif,
+                fontsize=10,
+                bbox=dict(facecolor="white", alpha=1),
+            )
+        ax.set_title(
+            f"FC-SC correlation within vs across subjects, {row['selected_rois']} rois"
+        )
+        plot_file = os.path.join(
+            boxplot_dir, f"withinvacross_FCSC_{row['selected_rois']}_box.png"
+        )
+        plt.savefig(plot_file, bbox_inches="tight")
+        plt.close()
+
+    # plot boxplot for FC-SC correlation but compare across different roi selections
+    # one plot for within subject and one for across subject
+    d = {
```

Comment:
Can you find a better name. using single letters that doe not correspond to a clear standard makes it error prone and hard to debug.